src/test/scala/com/microsoft/hyperspace/index/PartitionedDataHybridScanTest.scala

```diff
@@ -0,0 +1,103 @@
+
+class PartitionedDataHybridScanTest extends HybridScanTest with HyperspaceSuite {
+
+  override def beforeAll(): Unit = {
+    super[HyperspaceSuite].beforeAll()
+    import spark.implicits._
+    val dfFromSample = sampleData.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+    hyperspace = new Hyperspace(spark)
+    fileFormat = "parquet"
+    fileFormat2 = "json"
+
+    FileUtils.delete(new Path(sampleDataLocationRoot))
+    dfFromSample.write
+      .partitionBy("clicks", "Date")
+      .format(fileFormat)
+      .save(sampleDataFormatAppend)
+    dfFromSample.write.partitionBy("Date").format(fileFormat).save(sampleDataFormatAppend2)
+    dfFromSample.write.partitionBy("Date").format(fileFormat).save(sampleDataFormatDelete)
+    dfFromSample.write.partitionBy("imprs").format(fileFormat).save(sampleDataFormatDelete2)
+    dfFromSample.write
+      .partitionBy("Date", "imprs")
+      .format(fileFormat)
+      .save(sampleDataFormatDelete3)
+    dfFromSample.write.partitionBy("Date").format(fileFormat).save(sampleDataFormatDelete4)
+    dfFromSample.write.partitionBy("Date").format(fileFormat).save(sampleDataFormatBoth)
+    dfFromSample.write
+      .partitionBy("Date", "clicks")
+      .format(fileFormat2)
+      .save(sampleDataFormat2Append)
+    dfFromSample.write.partitionBy("Date").format(fileFormat2).save(sampleDataFormat2Delete)
+
+    val indexConfig1 = IndexConfig("indexType1", Seq("clicks"), Seq("query"))
+    val indexConfig2 = IndexConfig("indexType2", Seq("clicks"), Seq("Date"))
+
+    setupIndexAndChangeData(
+      fileFormat,
+      sampleDataFormatAppend,
+      indexConfig1.copy(indexName = "index_ParquetAppend"),
+      appendCnt = 1,
+      deleteCnt = 0)
+    setupIndexAndChangeData(
+      fileFormat,
+      sampleDataFormatAppend2,
+      indexConfig2.copy(indexName = "indexType2_ParquetAppend2"),
+      appendCnt = 1,
+      deleteCnt = 0)
+    setupIndexAndChangeData(
+      fileFormat2,
+      sampleDataFormat2Append,
+      indexConfig1.copy(indexName = "index_JsonAppend"),
+      appendCnt = 1,
+      deleteCnt = 0)
+
+    withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> "true") {
+      setupIndexAndChangeData(
+        fileFormat,
+        sampleDataFormatDelete,
+        indexConfig1.copy(indexName = "index_ParquetDelete"),
+        appendCnt = 0,
+        deleteCnt = 2)
+      setupIndexAndChangeData(
+        fileFormat,
+        sampleDataFormatDelete3,
+        indexConfig2.copy(indexName = "indexType2_ParquetDelete3"),
+        appendCnt = 0,
+        deleteCnt = 2)
+      setupIndexAndChangeData(
+        fileFormat,
+        sampleDataFormatBoth,
+        indexConfig1.copy(indexName = "index_ParquetBoth"),
+        appendCnt = 1,
+        deleteCnt = 1)
+      setupIndexAndChangeData(
+        fileFormat2,
+        sampleDataFormat2Delete,
+        indexConfig1.copy(indexName = "index_JsonDelete"),
+        appendCnt = 0,
+        deleteCnt = 2)
+    }
+  }
```

Comment:
Do we have a test that tests when a new partition is added? If it's combined with `HybridScanTest`, I suggest separating them out to better track.